File: App_Resources/Android/src/main/java/com/faceclaw/app/FaceclawDeviceInfoProbeListener.java
package com.faceclaw.app;

/** Callbacks from FaceclawDeviceInfoProbe to the TypeScript layer. */
public interface FaceclawDeviceInfoProbeListener {
    void onLog(String line);

    /** Lifecycle: connecting/authenticating for each arm, then querying. */
    void onState(String state, String detail);

    /** Terminal success: firmware versions and the firmware-extension string (empty on stock). */
    void onResult(String leftVersion, String rightVersion, String extension);

    /** Terminal failure (couldn't connect or read the version). */
    void onError(String message);
}

File: native/kotlin/shared/src/commonMain/kotlin/com/faceclaw/app/callbacks/FaceclawDeviceInfoProbeListener.kt
package com.faceclaw.app

/** Callbacks from FaceclawDeviceInfoProbe to the TypeScript layer. */
interface FaceclawDeviceInfoProbeListener {
    fun onLog(line: String?): Unit

    /** Lifecycle: connecting/authenticating for each arm, then querying. */
    fun onState(state: String?, detail: String?): Unit

    /** Terminal success: firmware versions and the firmware-extension string (empty on stock). */
    fun onResult(leftVersion: String?, rightVersion: String?, extension: String?): Unit

    /** Terminal failure (couldn't connect or read the version). */
    fun onError(message: String?): Unit
}
